README.md

```diff
@@ -28,15 +28,34 @@ Create a `PgLock.new` instance and call the `lock` method to ensure exclusive ex
 
 ```ruby
 PgLock.new(name: "all_your_base").lock do
-  if open?
-    API.log(circuit_breaker: true, enabled: true, service: service)
-  else
-    open!(error_count)
+  # stuff
+end
+```
+
+Now no matter how many times this code is executed across any number of machines, one block of code will be allowed to execute at a time.
+
+## Session based locking
+
+The postgres lock is unique across different database sessions, if the same session tries to aquire the same lock it will succeed. So while `PgLock` will guarantee unique execution across machines and processes, it will not block the same process (sharing the same connection session) from running. For example while you would think the middle block would not run in this example:
```

Comment:
Maybe mention the term`reentrant` here somewhere, since that's [the technical term](https://en.wikipedia.org/wiki/Reentrant_mutex) people may be expecting?